common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -470,6 +470,74 @@ static bool get_scan_results_from_nl_msg(sChannelScanResults &results, struct nl
     return true;
 }
 
+static std::shared_ptr<char> generate_client_assoc_event(const std::string &event, int vap_id,
+                                                         bool radio_5G)
+{
+    // TODO: Change to HAL objects
+    auto msg_buff = ALLOC_SMART_BUFFER(sizeof(sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION));
+    auto msg = reinterpret_cast<sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION *>(msg_buff.get());
+
+    if (!msg) {
+        LOG(FATAL) << "Memory allocation failed";
+        return nullptr;
+    }
+
+    // Initialize the message
+    memset(msg_buff.get(), 0, sizeof(sACTION_MONITOR_CLIENT_ASSOCIATED_NOTIFICATION));
+
+    char client_mac[MAC_ADDR_SIZE] = {0};
+    int supported_rates[16]        = {0}; // example: supported_rates=8c 12 98 24 b0 48 60 6c
+    int HT_MCS[16]                 = {0};
+    int16_t VHT_MCS[1]             = {0};
+    char ht_cap[8]                 = {0};
+    char ht_mcs[64]                = {0};
+    char vht_cap[16]               = {0};
+    char vht_mcs[24]               = {0};
+    int8_t max_tx_power            = 0;
```

Comment:
All of these are not required so it can be removed and also from the fieldsToParse.
just make sure the parser still work after you remove it.
If not, open an issue about it add a comment on the code about it with the issue number.

Reply:
Tried parsing without the additional fieldsToParse.
The `dwpal_string_to_struct_parse` function fails without them.